### ISSUE TITLE
Generalize ui:prop pattern

### ISSUE THREAD

**Issue by kav** [mozilla-services/react-jsonschema-form#372]
### Prerequisites

- [X] I have read the [documentation](https://github.com/mozilla-services/react-jsonschema-form/blob/master/README.md#readme);
- [X] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.

### Description
Right now we have several props for built in controls that are passed via uiSchema in 'ui:propName' format (disabled, readonly, autofocus, placeholder). When building custom widgets I'd love to pass other ui props to my widgets in a similar fashion.

```javascript
const disabled = Boolean(props.disabled || uiSchema["ui:disabled"]);
const readonly = Boolean(props.readonly || uiSchema["ui:readonly"]);
const autofocus = Boolean(props.autofocus || uiSchema["ui:autofocus"]);
```
[in code](https://github.com/mozilla-services/react-jsonschema-form/blob/master/src/components/fields/SchemaField.js#L141)

As before; will PR

Implementation question if this makes sense: is there value to the uiSchema props having the `ui:` prefix? They are already in uiSchema so it's clearly UI related. Happy to implement either way but wanted to start that conversation as well.

### Version: 0.40.0

**Comment by kav** [comment]
Also how does this high level pattern relate to a `ui:options` as a bit of a grab bag. Aren't these all options of a sort? Should we perhaps generalize there instead?

**Comment by n1k0** [comment]
See #370 which is related. So there's already a way to specify [custom widget options](https://github.com/mozilla-services/react-jsonschema-form#custom-widget-options), though I'm debating we move to supporting `ui:options` only. We may also go further and move everything to `ui:options` as suggested though I'm not entirely sure just yet. Please chime in #370 if you have thoughts about all that. 

> is there value to the uiSchema props having the `ui:` prefix? They are already in uiSchema so it's clearly UI related

Not sure to properly understand the question, but the reasoning is to disambiguate ui directives from field names; think of `ui:` as a JSON namespace if that makes sense.

**Comment by n1k0** [comment]
Solved by #378 (#370).
